app/src/behavior.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <sys/util.h>
+#include <logging/log.h>
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#include <zmk/matrix.h>
+#include <zmk/sensors.h>
+#include <zmk/keymap.h>
+#include <drivers/behavior.h>
+#include <zmk/behavior.h>
+
+#include <zmk/event-manager.h>
+#include <zmk/events/position-state-changed.h>
+#include <zmk/events/behavior-state-changed.h>
+#include <zmk/events/sensor-event.h>
+
+int zmk_behavior_state_changed(const struct behavior_state_changed *ev) {
+    int ret;
+    if (ev->pressed) {
+        ret = behavior_keymap_binding_pressed(ev);
+    } else {
+        ret = behavior_keymap_binding_released(ev);
+    }
+    if (ret < 0) {
+        return ret;
+    }
+    switch (ret) {
+    case ZMK_BEHAVIOR_TRANSPARENT:
+        return ZMK_EV_EVENT_BUBBLE;
+    case ZMK_BEHAVIOR_OPAQUE:
+        return ZMK_EV_EVENT_HANDLED;
+    case ZMK_BEHAVIOR_CAPTURED:
+        return ZMK_EV_EVENT_CAPTURED;
```

Comment:
This mapping is a little ugly, but the zero (0) return value for events is 'bubble' and the zero return value for behaviors is 'opaque'. The code of behaviors and events assumes these return values often when returning output from other functions.